### ISSUE TITLE
Request for passphrase with visible stdin

### ISSUE THREAD

**Issue by vkuznet** [xwmx/nb#35]
Hi,
I found that in certain cases nb asks for my passphrase when performing an action. For instance I added a bookmark and here is what I saw in my terminal:
```
nb https://www.miek.nl/go/ "Go book"
! Unable to download page at https://www.miek.nl/go/
Added: [12] 20201011123927.bookmark.md "(www.miek.nl)"
my-shell-prompt >
Enter passphrase for key '/Users/vk/.ssh/rsa':
```
If I type the passphrase it will be visible on terminal, if I abort everything is fine and bookmark is created. There are few issues here:
- request for any kind of password/passphrase should not be echoed to terminal
- it is unclear why it asks my passphrase for certain repo, e.g. I have passphrase for bitbucket, for github, etc.
- and, it is unclear why that even when I abort (through Ctrl-C) the action has finished correctly. I would expect that abort should abort the action and not complete it.

Thanks,
Valentin.

**Comment by xwmx** [comment]
I think you should be able to resolve this using ssh-agent:

https://docs.github.com/en/free-pro-team@latest/github/authenticating-to-github/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent

Please let me know if that works.

I don't know why the prompt is in the clear. Currently this operation runs the background, so that might be influencing it.

I'm still looking into this...

**Comment by vkuznet** [comment]
I added my keys/passphrases to ssh agent and verified that they appear in my keychain, but the problem still remains. upon adding or deleting bookmark I see prompt for passphrase. And, for completeness, I'm using macOS with zsh and iterm2.

**Comment by vkuznet** [comment]
Another piece of information. I found empirically that it only happens when I'm in git repository, that's explain a passphrase prompt. When I am not in github repository area, then I don't have this issue, i.e. I can add bookmark and nb does not ask me for passphrase. Therefore, I think the problem relies somewhere which relates to git area.

**Comment by xwmx** [comment]
Interesting. Thanks for the information. Git operations in `nb` are generally scoped to the notebook using the `git -C` option and sounds like there might be an unscoped git operation happening in the current working directory. I'll have to review the git operations to determine what's going on. I haven't been able to reproduce the issue yet (also macOS, zsh, iterm2).

**Comment by xwmx** [comment]
The prompts, documentation, error handling, and test coverage around this have been greatly improved, so this issue should be resolved. Prompts should now be fully suppressed during automatic syncing.

These changes are available as of version 5.7.0.

Thanks for the heads up! Please let me know if you encounter continued issues with this.
